feat: 설정 화면 페이지 UI 구현

devjaeyoon opened the pull request:
### ✨ 이슈 번호

- #13 

### 📌 설명

설정 페이지의 UI 작업을 진행한 Pull request입니다.

### 📃 작업 사항

- [x]  상단 설정 타이틀
- [x]  컨트롤러를 구성한다.
    - [x]  광고 감지 토글
    - [x]  기존 채널 이동 토글
- [x] TabBar 좌우 패딩 제거
- [x] ToggleButton 크기 축소

### 💡 참고 사항

설정 페이지 UI 구현을 진행하며 공용 컴포넌트인 TabBar와 ToggleButton에 수정사항이 생겼습니다.
TabBar에 빈 공간이 생겨 좌우 패딩을 제거함으로써 빈 공간을 제거했고, ToggleButton을 사용하며 생각보다 크다 판단했기에
@Eun0713 님과 상의하여 크기를 축소했습니다. 이 점 참고 부탁드립니다.

모바일 환경에서 UI는 아래와 같으며 최상단 흰 공간은 추후 작업할 header가 차지할 영역입니다.

![image](https://github.com/user-attachments/assets/921e687b-95bd-4510-b1dd-0309226bddb1)

### 💭 리뷰 사항 반영

- [x] 동일한 li 구조 SettingListitem 컴포넌트로 분리 및 리팩토링
- [x] section 태그 div로 변경
- [x] strong 태그 h2 태그로 변경
- [x] settingList에서 동일 타입 다른 타입으로 변경

### ✅ Pull Request 체크 사항

- [x] 가장 최신 브랜치를 pull 하였습니다.
- [x] base 브랜치명을 확인하였습니다.
- [x] 코드 컨벤션을 모두 확인하였습니다.
- [x] 브랜치명을 확인하였습니다.

devjaeyoon committed:
style: TabBar 컴포넌트에서 좌우 패딩 제거

기존 px-4 속성을 제거하여 양쪽 여백 없이 전체 너비로 탭이 배치되도록 수정

devjaeyoon committed:
style: ToggleButton 컴포넌트 크기 축소

기존 크기가 다소 컸기에 높이와 너비를 소폭 축소함

devjaeyoon committed:
feat: 설정 페이지 UI 구현

- 광고 감지 및 기존 채널 복귀 기능용 토글 UI 추가
- TabBar 및 ToggleButton 컴포넌트 적용

faber-h commented on src/pages/Settings.jsx:
```diff
@@ -1,5 +1,42 @@
+          <li className="my-4 flex h-[88px] w-full min-w-80 rounded-md bg-gray-100 p-2">
+            <section className="w-72">
+              <strong className="text-lg">기존 채널로 이동</strong>
+              <p className="mb-[-4px] text-[#888888]">
```
하나의 문장을 임의로 나누신 이유가 궁금합니다~

devjaeyoon replied:
`이전 채널의 광고가 종료되면`과 `자동으로 복귀합니다.`를 나눈 이유는 이 두 개를 하나의 `<p>`로 합치면 핸드폰 환경마다 텍스트의 줄바꿈이 다르게 될 것 같아(예를 들어 아이폰 미니는 줄바꿈이 되는데 아이폰 맥스는 줄바꿈되지 않는) 모든 환경에서 동일하게 텍스트를 보여줄 수 있도록 줄바꿈을 진행했습니다. 또한, 첫 번째 문장에 `margin-top`, `margin-bottom`에 별도의 스타일링이 주어져야 하기 때문에 임의로 나눴습니다.

faber-h commented on src/pages/Settings.jsx:
```diff
@@ -1,5 +1,42 @@
+      <TabBar />
+      <div className="p-4">
+        <ul>
+          <li className="my-4 flex h-[88px] w-full min-w-80 rounded-md bg-gray-100 p-2">
```
`li` 구조가 비슷한거같은데 컴포넌트로 분리하면 좋을 것 같습니다~

devjaeyoon replied:
분리하는게 더 깔끔할 것 같네요. 좋은 피드백 감사드립니다!

faber-h commented on src/pages/Settings.jsx:
```diff
@@ -1,5 +1,42 @@
+      <div className="p-4">
+        <ul>
+          <li className="my-4 flex h-[88px] w-full min-w-80 rounded-md bg-gray-100 p-2">
+            <section className="w-72">
```
section을 사용하신 이유가 있을까요?
컨테이너로 사용하신 것 같아 보이는데 만약 일반 컨테이너로 사용한다면 `<div>` 태그가 좋을 것 같습니다~

[`HTML <section> 태그 – 올바른 이해와 사용 방법`](https://codingeverybody.kr/html-section-%ED%83%9C%EA%B7%B8-%EC%98%AC%EB%B0%94%EB%A5%B8-%EC%9D%B4%ED%95%B4%EC%99%80-%EC%82%AC%EC%9A%A9-%EB%B0%A9%EB%B2%95/)

devjaeyoon replied:
`div`의 남용을 방지하고자 `section` 태그를 사용했는데 `section`은 문서의 구조 일부로 간주할 수 있고 의미상 독립된 내용의 묶음일 때만 `section`을 사용하라고 하네요. 레이아웃 구분은 말씀하신대로 `<div>`가 맞는 것 같습니다. 좋은 리뷰 감사드립니다!

d0-neii commented on src/pages/Settings.jsx:
```diff
@@ -1,5 +1,42 @@
+        <ul>
+          <li className="my-4 flex h-[88px] w-full min-w-80 rounded-md bg-gray-100 p-2">
+            <section className="w-72">
+              <strong className="text-lg">광고 감지</strong>
```
`strong`보다는 섹션 제목 역할이라 `h2`를 쓰는 건 어떨까요?
섹션 제목이라면 `h2` 태그를 쓰는게 스크린 리더 사용자분들께 더 도움이 될 거 같아서요!

devjaeyoon replied:
`h2`를 쓰려면 `h1`이 반드시 선행되어야 하는 줄 알았는데 맥락이 서브 섹션의 제목이니 `h2`가 더 좋은 것 같네요. 좋은 리뷰 감사드립니다!

devjaeyoon committed:
style: 토글 버튼 width 수정

devjaeyoon committed:
refactor: 설정 항목 컴포넌트 분리 및 반복 렌더링 구조로 리팩토링

- 중복된 설정 항목 UI를 SettingListItem 컴포넌트로 분리
- 설정 정보를 배열로 추상화하여 map 기반 렌더링 구현

faber-h commented on src/pages/Settings.jsx:
```diff
@@ -1,5 +1,39 @@
+
+const settingList = [
+  {
+    type: "isAdDetect",
```
settingList로 잘 묶어주셨네요!
다만 type 필드가 현재 모든 항목에서 동일한 값을 가지고 있어 다른 키 값을 사용해주시면 좋을 것 같습니다~

devjaeyoon replied:
리팩토링 하는 과정에서 무심코 넘어갔었네요! 좋은 리뷰 감사드립니다~

devjaeyoon committed:
refactor: settingList의 type 필드를 고유하게 수정

- 동일한 type 값을 가지던 항목을 isAdDetect, isReturnChannel로 구분
- map 렌더링 시 key 충돌 방지 및 추후 기능 확장성 고려

Eun0713 reviewed (approved):
LGTM🍀

faber-h reviewed (approved):
LGTM🔥
